hyperspy/learn/mva.py

```diff
@@ -1317,22 +1369,666 @@ def _estimate_elbow_position(self, curve_values):
+                         use_decomposition_for_centers=False,
+                         number_pca_components=None,
+                         navigation_mask=None,
+                         signal_mask=None,
+                         algorithm='kmeans',
+                         return_info=False,
+                         **kwargs):
+        """
+        Cluster analysis of a signal
+
+        Results are stored in `learning_results`.
+
+        Parameters
+        ----------
+        n_clusters : int
+            Number of clusters to find.
+        scaling : {"standard","norm","minmax", or scikit learn scaling method} 
+            default: 'standard' 
+            Preprocessing the data before cluster analysis requires scaling 
+            the data to be clustered to similar scales. Standard scaling
+            adjusts each feature to have uniform variation. Norm scaling 
+            adjusts treats the set of features like a vector and 
+            each measurement is scaled to length 1. 
+            You can also pass one of the scikit-learn preprocessing 
+            scale_method = import sklearn.processing.StandadScaler
+            scaling = scale_method
+            See scaling methods in scikit-learn preprocessing for further 
+            details.        
+        scaling_kwargs : dict
+            Additional parameters passed to the cluster scaling algorithm. 
+            See sklearn.preprocessing scaling methods for further details
+        use_decomposition_results : bool
+            If True (recommended) the signal's decomposition results are used
+            for clustering. Note - If this option is not used the raw data
+            is used. This can be memory intensive and is only recommened if
+            the Signal has a small `signal_dimension`.
+        use_decomposition_for_centers : bool
+            If True (recommended) the pca results are used for building the 
+            the cluster centers from the clustered label results. 
+            If False the original signal data is used.
+        number_pca_components : int, default None
+            If you are getting the cluster centers using the decomposition 
+            results (use_decomposition_for_centers=True) you can define how 
+            many PCA components to use. If set to None the method uses the 
+            estimate of significant components found in the decomposition step
+            using the elbow method and stored in the 
+            ``learning_results.number_significant_components`` attribute.
+        navigation_mask : boolean numpy array
+            The navigation locations marked as True are not used in the
+            decomposition.
+        signal_mask : boolean numpy array
+            The signal locations marked as True are not used in the
+            decomposition.
+        algorithm : { "kmeans" | "agglomerative" }
+            See scikit-lear documentation. Default "kmeans"
+        **kwargs: dict  optional, default - empty
```

Comment:
`**kwargs : dict` single space before the `:`.

Reply:
fixed